src/comm/ModDevPoll.cc

```diff
@@ -48,9 +48,8 @@
 
 #define DEBUG_DEVPOLL 0
 
-// OPEN_MAX is defined in <climits>
-#define DEVPOLL_UPDATESIZE  OPEN_MAX
-#define DEVPOLL_QUERYSIZE   OPEN_MAX
+#define DEVPOLL_UPDATESIZE  Squid_MaxFD
+#define DEVPOLL_QUERYSIZE   Squid_MaxFD
```

Comment:
We should get rid of these non-standard (and potentially clashing) macros completely and use Squid_MaxFD in ModDevPoll code instead.

```suggestion
```

To avoid code duplication, the lines using these values should be reordered so that each code snippet is using Squid_MaxFD exactly once. For example:

```C++
    dpoll_nfds = Squid_MaxFD;
    do_poll.dp_fds = (struct pollfd *)xcalloc(
                         dpoll_nfds, sizeof(struct pollfd)
                     );
```

and

```C++
    devpoll_update.size = Squid_MaxFD;
    devpoll_update.pfds = (struct pollfd *)xcalloc(
                              devpoll_update.size, sizeof(struct pollfd)
                          );
```

If both snippets should use the same number of descriptors, then the second snippet should become

```C++
    devpoll_update.size = dpoll_nfds;
    devpoll_update.pfds = (struct pollfd *)xcalloc(
                              devpoll_update.size, sizeof(struct pollfd)
                          );
```

Reply:
While the original code used the same value for both, it is not required to do so.  Both must be no larger than the current RLIMIT_NOFILES, since you can't request to poll on more file descriptors than you can possibly open, nor receive more poll responses than the number of file descriptors that are currently open.

DEVPOLL_UPDATESIZE is the number of fd's that can fit in the buffer used to batch up requests to add/remove fd's from the list /dev/poll is monitoring.  Having it be Squid_MAXFD allows requesting to add or remove every fd at once, and is likely overkill - perhaps this should be a fixed value such as 128 or 256 (the old OPEN_MAX value).  The cost of having it lower is simply making more write() system calls if a list of fd's larger than the buffer size is being presented all at once.

DEVPOLL_QUERYSIZE is the maximum number of fd's that can be returned in a single call to the DP_POLL ioctl() [the equivalent of a poll() or select() system call in this backend]. Having it be Squid_MAXFD allows being notified of a result for every fd at once, and may also be overkill, but I don't know what value to suggest here.

Reply:
This change request is outdated. I split/moved the discussion to https://github.com/squid-cache/squid/pull/1551#discussion_r1375553125 and https://github.com/squid-cache/squid/pull/1551#discussion_r1375554045. I am resolving this change request.